src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipIncorporationService.java

```diff
@@ -1,40 +1,79 @@
 package uk.gov.companieshouse.limitedpartnershipsapi.service;
 
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.FILING_KIND_REGISTRATION;
 import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.LINK_SELF;
 import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.URL_GET_INCORPORATION;
 
 import java.time.LocalDateTime;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.model.transaction.Resource;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.limitedpartnershipsapi.exception.ServiceException;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dao.IncorporationDataDao;
 import uk.gov.companieshouse.limitedpartnershipsapi.model.dao.LimitedPartnershipIncorporationDao;
 import uk.gov.companieshouse.limitedpartnershipsapi.repository.LimitedPartnershipIncorporationRepository;
 
+
 @Service
 public class LimitedPartnershipIncorporationService {
 
-    public static final String LIMITED_PARTNERSHIP_REGISTRATION_KIND = "limited-partnership-registration";
     private final LimitedPartnershipIncorporationRepository repository;
+    private final TransactionService transactionService;
 
     public LimitedPartnershipIncorporationService(
-            LimitedPartnershipIncorporationRepository repository) {
+            LimitedPartnershipIncorporationRepository repository,
+            TransactionService transactionService) {
         this.repository = repository;
+        this.transactionService = transactionService;
     }
 
-    public String createIncorporation(String userId, String transaction) {
-        var dao = new LimitedPartnershipIncorporationDao();
-        dao.getData().setKind(LIMITED_PARTNERSHIP_REGISTRATION_KIND);
-        // TODO set etag
-        dao.setCreatedAt(LocalDateTime.now());
-        dao.setCreatedBy(userId);
+    public String createIncorporation(Transaction transaction, String requestId, String userId)
+            throws ServiceException {
+        var incorporationDataDao = new IncorporationDataDao.Builder()
```

Comment:
Not simpler to ditch the builder and just have one new line?

`dao.getData().setEtag(GenerateEtagUtil.generateEtag());`

Reply:
Possibly but it was starting to get a bit ugly so I used a builder.

Reply:
Removed. See above comment.